project_task_subtask_same_project/models/project_task.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, _
+from odoo.exceptions import ValidationError
+
+
+class ProjectTaskSubtaskSameProject(models.Model):
+    _inherit = "project.task"
+
+    def _check_subtask_not_in_different_project(self):
```

Comment:
Hello @rivo2302 
It would be preferable to use `@api.constrains` here instead of calling the check in the `write()` method.